C/services/south/ingest.cpp

```diff
@@ -499,14 +503,35 @@ void Ingest::processQueue()
 				m_failCnt = 0;
 				std::map<std::string, int>		statsEntriesCurrQueue;
 				AssetTracker *tracker = AssetTracker::getAssetTracker();
+				StorageAssetTracker *satracker = StorageAssetTracker::getStorageAssetTracker();
+				if ( satracker == nullptr)
+                                {
+                                        Logger::getLogger()->error("%s:%d could not initialize satracker ", __FILE__, __LINE__);
+					return;
+                                }
 
 				string lastAsset = "";
 				int *lastStat = NULL;
+				std::map <std::string , std::set<std::string> > assetDatapointMap;
+
 				for (vector<Reading *>::iterator it = q->begin();
 							 it != q->end(); ++it)
 				{
 					Reading *reading = *it;
 					string assetName = reading->getAssetName();
+                                        const std::vector<Datapoint *> dpVec = reading->getReadingData();
+					std::string temp;
+
+                                        for ( auto dp : dpVec)
+                                        {
```

Comment:
I think we should maintain DPs separately here rather than combining them in a single string to take care of change in order of DPs.

Reply:
have modified data structure